enhancements/last-scanned-timestamps.md

```diff
@@ -0,0 +1,87 @@
+---
+title: last-scanned-timestamps
+authors:
+  - TBD
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @rhmdnd
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2024-04-10
+last-updated: 2024-04-10
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - /
+replaces:
+  - /
+superseded-by:
+  - /
+---
+
+# Add `last_scanned_timestamp` to `compliancecheckresults.compliance.openshift.io` objects
+
+## Summary
+
+ACS implements an integration with Compliance Operator. For this, ACS reports when a scan was completed. Currently
+Compliance Operator does not update results when a scan did not changed, this makes it impossible for ACS to determenistically
+indicate a completed scan (e.g. to send scan reports to users automatically).
+
+## Motivation
+
+Identify deterministically when a scan is completed.
+
+### Goals
+
+Goal is to identify when a `profiles.compliance.openshift.io`, `compliancesuites.compliance.openshift.io` or `compliancescans.compliance.openshift.io` is completed.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+Add a `LastTimeStamp` field to the `ComplianceCheckResult` which is updated after every scan.
```

Comment:
I just realized we have a similar field already on the `ComplianceSuite` and `ComplianceScan` objects:

```
╭─lbragstad@p1 ~
╰─➤  $ oc get scans -o json ocp4-cis | jq .status.conditions
[
  {
    "lastTransitionTime": "2024-04-12T20:21:52Z",
    "message": "Compliance scan run is done running the scans",
    "reason": "NotRunning",
    "status": "False",
    "type": "Processing"
  },
  {
    "lastTransitionTime": "2024-04-12T20:21:52Z",
    "message": "Compliance scan run is done and has results",
    "reason": "Done",
    "status": "True",
    "type": "Ready"
  }
]
╭─lbragstad@p1 ~
╰─➤  $ oc get suites -o json cis | jq .status.conditions
[
  {
    "lastTransitionTime": "2024-04-12T20:21:52Z",
    "message": "Compliance suite run is done running the scans",
    "reason": "NotRunning",
    "status": "False",
    "type": "Processing"
  },
  {
    "lastTransitionTime": "2024-04-12T20:21:52Z",
    "message": "Compliance suite run is done and has results",
    "reason": "Done",
    "status": "True",
    "type": "Ready"
  }
]
```

To keep things consistent, I wonder if we can just implement status conditions on `ComplianceCheckResults`.